enhancements/installer/component-selection.md

```diff
@@ -94,233 +95,183 @@ N/A - no api extensions are being introduced in this proposal
+  For a 4.11 cluster, this matches `v4.11`.  We may add or remove capabilities from this set in 4.12 clusters.
 
-From the remaining set of resources:
-3) if any resource is part of an explicitly excluded capability, remove it
-4) if any resource is part of an explicitly included capabilitity, keep it
-5) if the inclusionDefault is Exclude, remove all remaining resources(i.e. not covered by (4)) that are part of a capability
+Admins who want to pre-emptively exclude future capabilities can use `None` or `v4.11` or another `baselineCapabilitySet` which will not evolve with time.
 
+We may decide to add additional sets like `v4.12` or `myGreatUseCase` in the future, but do not commit to doing so.
 
-The `inclusionDefault` field (carried through to the CVO) allows an Admin to choose what behavior they want for new
-capabilities that might come as part of a future upgrade (either have the new capability installed during
-the upgrade, or exclude it).  The `inclusionDefault` field itself will default to `Include`.
+#### Manifest annotations
 
-Specifying the same capability as an `include` and an `exclude` is an error condition.
+Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
 
-Specifying a capability name that isn't represented in the payload is an error condition.  Admins who want to
-pre-emptively exclude future capabilities can use `inclusionDefault: Exclude` and then explicitly include
-anything they want the cluster to have.
+```yaml
+annotations:
+  capability.openshift.io/name: console
+```
 
-Examples of candidate components to be treated as capabilities:
+If a resource participates in multiple capabilities, it should specify all capabilities as plus-sign separated
+values in the annotation(e.g. `capability.openshift.io/name=console+monitoring`).  The resource will only be 
+included if all the capabilities on the resource are enabled.  This allows a component
+to define a `ServiceMonitor` which is part of both the `monitoring` capability and the `Foo` capability.  If
+either of those capabilities are excluded, the `ServiceMonitor` should not be created.  (If `monitoring` is excluded
+then no ServiceMonitors should be created, if only `Foo` is excluded, then only Foo's ServiceMonitors should be
+excluded while ServiceMonitors associated with other capabilities would not be).
 
-* console
-* imageregistry
-* samples
-* cluster baremetal operator
-* olm/marketplace
-* csi-*
-* insights
-* monitoring
-* ???
+To determine which manifests are managed for a given cluster, the CVO will:
 
+1. Start with all the manifests in the payload.
+2. Remove all the manifests that have no annotations.
+3. If an exclusion identifier is set, remove all the manifests with the exclusion annotation `exclude.release.openshift.io/{exclusionID}=true`.
```

Comment:
A bit of this happened as a short-term IBM-y hack openshift/cluster-version-operator#269, without going through the enhancement process (possibly from before we had an enhancement process), and we've been [dragging it around since then][1] as undocumented cruft.  Would be good to drop it if possible.  I'm also fine dropping this line here to keep the cruft less documented ;)

[1]: https://github.com/openshift/library-go/blob/f2c20ff2ad2771dd39c1482658f18c2587da6469/pkg/manifest/manifest.go#L128-L133

Reply:
oh, right this is the other exclusion stuff....   ok.  i dunno how many people will read this and think the "exclusion identifier" refers to capability exclusion, like I did.

might be worth a clarification.

Reply:
I've just dropped this line with aea25fe1 -> 5f94ef77.  It's old, niche cruft, which we may not even care about anymore, and a larger explanation seemed like it would be distracting without adding significant value to folks looking to understand how caps work.  The presence of cluster-profile and tech-preview filtering is enough to get the "there are other types of manifest filters around", and folks don't need to get caught up in the details of all of those mechanisms here.

Developers who are interested in all the knobs we're supporting in manifests should be looking [here][1], and not just in this subsection of the caps proposal.

[1]: https://github.com/openshift/enhancements/blob/e6be674880cb8361e1bb45bcba6b4ed753cbbac1/dev-guide/cluster-version-operator/dev/operators.md#manifest-inclusion-annotations